### ISSUE TITLE
Add categories to the "What kind of building is this?" Quest

### ISSUE THREAD

**Issue by RiffLord** [westnordost/StreetComplete#2177]
**Use case**

Some areas have many abandoned buildings and/or ruins which can't be adequately classified by the categories available in the _What kind of building is this?_ quest. I believe it would be useful to be able to mark these buildings as abandoned.

On a related note, I believe the quest would also benefit from having a category for historical buildings which could encompass used, unused, public or private buildings of historical relevance which otherwise wouldn't fit into any of the currently available categories. While many historical buildings are used for purposes which can be classified with the options available (Civic, Religious or Commercial buildings) there are some which don't fall into these categories and might otherwise remain unmarked.

**Proposed Solution**

Add an _Abandoned building/Ruin_ and a _Historical building_ option to the _Other_ category in the options for the quest.

Something like this (the icons were repurposed from those available just as a placeholder):

![Untitled](https://user-images.githubusercontent.com/47884610/96376329-8df30e80-117e-11eb-9a92-cf684be117c0.jpg)

**Comment by matkoniecz** [comment]
Note that abandoned bunker should still be answered as "bunker", abandoned shed as "shed" and so on.

Historic buildings answer - which tag would be used in this case?

**Comment by RiffLord** [comment]
Regarding generic abandoned buildings, what about other categories which might cause confusion? 
For example, in the area where I live there is the ruin of a restaurant. If I were to classify it as a Shop per the app's guidelines, wouldn't it show up as a result along with other commercial activities for people using OSM to find services in the area?

As for the historic buildings, I assume that simply using a _Historic_ tag would be too vague perhaps?
In that case what about creating a separate Historic category, including tags such as _Castle_, _Fort_, _Watchtower_, _Observation Post_, etc?

The reasoning behind this is that while all over the world buildings such as these are often museums or tourist attractions and can be tagged appropriately through the app, what about the similar places which don't have this function?
How do we tag those historic buildings which remain officially unused but accessible to people?

**Comment by westnordost** [comment]
It would be possible to tag something like `abandoned=yes` or `historic=yes`. These tags are even well in use.

**Comment by RiffLord** [comment]
So these characteristics should be specified by editing tags after the fact on OpenStreetMaps rather than adding options to the quest in the StreetComplete app?

**Comment by westnordost** [comment]
No, on the contrary! What I meant to say is that it would be quite possible to add these two options in streetcomplete because there is an accepted tagging for that

Am 19. Oktober 2020 02:41:33 MESZ schrieb Bruno Pezer <notifications@github.com>:
>So these characteristics should be specified by editing tags after the
>fact on OpenStreetMaps rather than adding options to the quest in the
>StreetComplete app?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/2177#issuecomment-711450989

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by westnordost** [comment]
>  For example, in the area where I live there is the ruin of a restaurant. If I were to classify it as a Shop per the app's guidelines, wouldn't it show up as a result along with other commercial activities for people using OSM to find services in the area?

No, because this quest is just about the building type, ot the amenity within.

**Comment by RiffLord** [comment]
> 
> As for the historic buildings, I assume that simply using a _Historic_ tag would be too vague perhaps?
> In that case what about creating a separate Historic category, including tags such as _Castle_, _Fort_, _Watchtower_, _Observation Post_, etc?
> 

I understand. In this case I'd say it makes little sense to add an _Abandoned_ tag to this quest. What about _Historic_? Is it too generic? Would the quest benefit from a new category for these buildings with tags as suggested above?

**Comment by westnordost** [comment]
Did you just cite yourself?

Anyway, maybe a historic category would make sense. But just "historic building" would be fine too in my opinion. What exactly do you have in mind for possible selections in this category?

**Comment by RiffLord** [comment]
Yeah, as a reminder of the ideas I had for the selections which could be a part of the possible historic category. 
However if you feel that a generic Historic building tag is enough, I've already added it to my fork of the app, in the "Other" section. I could test to see if everything works as expected.
Is there anything I should do or be aware of before making a pull request?

**Comment by westnordost** [comment]
>  I've already added it to my fork of the app

Nice!

>  Is there anything I should do or be aware of before making a pull request?

Well, a historic building should be tagged with `building=*` + `historic=building` (or `=yes`). An abandoned building with building=*` + `abandoned=yes`. Ruins with `ruins=yes`. Do you want to include all three options?
Any idea for fitting icons?

**Comment by RiffLord** [comment]
I'd be glad to work on including all of the mentioned options. I'll take some time to study the code further and find out how to approach tagging these buildings. As of this moment, if I tag a building as Historic from the app, the result on OSM is `building=historic`, so it classifies the building as such rather than adding a "historic" tag.

I think a historic building could use the same icon as Civic and Government:

![Untitled](https://user-images.githubusercontent.com/47884610/96471027-306aca80-122f-11eb-8e03-fc1b98d29dd0.png)

**Comment by westnordost** [comment]
Don't worry about the icons, I will draw them after your PR is merged. I was just asking for ideas.

Tagging it correctly will not be much effort, just look at how it is done with the man_made=storage_tank.

**Comment by rhhsm** [comment]
Lots of disused and abandoned buildings where I'm mapping, but I don't think SC is suitable for tagging them. I think a building should first of all be tagged with the type it was built for, and then an abandoned or disused tag can be added if it's no longer used. If we allow a building to be tagged as abandoned using SC, it would mean no quests would appear any more asking if it was a house, church, office, etc. isn't it? I think that is worse than not tagging it. A separate quest asking of every building if it might be disused or abandoned would be a lot of spam. I can imagine it could be an additional question to the building type quest, but again the answer will almost always be no (I hope). I think the historic tag has the same problem: it's a secondary characteristic of a building. Any building should first of all be tagged for what it was built for (not what it is used for), and then maybe for its historic value.

**Comment by westnordost** [comment]
@rhhsm I think SC is very much suitable for tagging them. SC is a tool for people doing ground survey. Where else other than on a survey is it possible to determine whether a building is ruined or abandoned?

But I understand your point of view: the property (abandoned or not) is separate from `building=*` and thus should be recorded separately. But since it is not appropriate to ask for every single building if it is abandoned or not, your verdict is, that this is not a property that should be recorded with StreetComplete.

I see it from a difference perspective, from the perspective of a surveyor (a user of the app): If the building is abandoned or even ruined, or it is just some kind of historic building, it is quite impossible for the anyone on-site to find out what this building was created for. It's just "some building like any other". Because it is not well possible to answer this, the quest is not asked for abandoned buildings, ruined buildings, historic buildings, military buildings, buildings that are (tourist) attractions, buildings tagged with `power` etc.

So, from a surveyor perspective, it very much makes sense to answer "(I can't answer this because) the building is abandoned". Without this option, the user would not be able to give an answer to this case, but the app should enable the user to do so. Since the quest just generally asks for "what kind of building this is?", it makes sense to put this answer option amongst all the others instead of into the special "other answers...." menu.

Regarding your point that tagging `ruins=yes` (etc.) leads to that the quest is not shown (for other users) any more: Yes, that's right, but this is exactly the point. *If someone on site cannot determine what kind of building it is because it is abandoned, then it makes sense to not ask every passing user again about it*. Just like for `opening_hours:signed=no`. That doesn't mean that anyone else with knowledge from another source cannot set the building type later, just not with StreetComplete.

**Comment by rhhsm** [comment]
I'm in favour of having an answer option "Can't say because the building is so damaged that its function cannot be determined", but then the tag to add is ruins:building=yes, as "abandoned" means that "Such features will still have some physical form reflective of their former use visible in the landscape." https://wiki.openstreetmap.org/wiki/Key:abandoned: If their former use is still visible, it should be tagged for it.

**Comment by westnordost** [comment]
> If their former use is still visible, it should be tagged for it.

Sure, but why wouldn't people do that even if there is such an "abandoned" option? What building type would you select if you see an apartment building which seems to be deserted? Which one is the defining property of a building - it being an apartment house or it being some kind of deserted building is in the eye of the beholder. The "openstreetmap tags" categorization can't be forced on users if it does not coincide with how in reality the building is perceived.

**Comment by rhhsm** [comment]
I'd definitely tag it as an apartment building with SC, and then use Vespucci or iD to add a disused or abandoned tag. I believe that the consensus on OSM is that buildings should first of all be tagged with what they were built for, and then in addition, if applicable, for what they are used for and for their lifecycle status. I don't think SC should tempt mappers to deviate from the OSM consensus.

**Comment by westnordost** [comment]
StreetComplete should enable mappers to map what they see.

**Comment by mnalis** [comment]
@rhhsm I agree with the part is USAGE differs, eg. something was built as a church, and later transformed to pub, would be `building=church` and `building:use=retail`. However, while current use (`building:use`) is almost always easily known,  original purpose is in many cases hard to know on-site (although there are exceptions like church or warehouse where it is much easier to correctly guess).

However, `building=ruins` / `abandoned=yes` and similar tags are often special, as it might not even be possible to discern original purpose (especially for very old, or razed to the ground in war and never rebuilt - as is often the case in parts of my country). 

And even it was possible, something that is basically a pile of collapsed bricks should IMHO not be tagged just as `building=yes` or `building=house`, with hope that someone will later remember to mark it as ruins/abandoned in desktop editor.

I would think that marking it as `building=ruins` or similar would be preferable in such situations.

**Comment by rhhsm** [comment]
I think there's a lot of confusion here about the meaning of abandoned: Please read https://wiki.openstreetmap.org/wiki/Key:abandoned: The defining difference between an abandoned building and a ruined one, is that the former purpose of an abandoned building can still be recognised, and that of a ruined building not. If a house in a very bad state of repair can still be recognised as a house, it should be tagged abandoned:building=house. If it cannot be recognised, it should be tagged ruins:building=yes
Also note the taginfo counts here https://wiki.openstreetmap.org/wiki/Lifecycle_prefix (ruins: and ruined: are seldom used).

**Comment by westnordost** [comment]
I can very well imagine a generic building of unknown purpose that is clearly abandoned (windows are nailed shut with chipboard) that I'd not at all classify as ruin. The ruins of a church for example will be recognizable as the ruins of a church. See for example Nikolaikirche in Hamburg. Clearly a ruin,  clearly revognizable as a church and not at all abandoned.

So whereever you got that definition of abandoned vs ruined from, it doesn't seem to coincide with reality.

By the way , i am talking about setting abandoned=yes, ruins=yes, historic=yes

Am 26. Oktober 2020 08:04:33 MEZ schrieb Richard <notifications@github.com>:
>I think there's a lot of confusion here about the meaning of abandoned:
>Please read https://wiki.openstreetmap.org/wiki/Key:abandoned: The
>defining difference between an abandoned building and a ruined one, is
>that the former purpose of an abandoned building can still be
>recognised, and that of a ruined building not. If a house in a very bad
>state of repair can still be recognised as a house, it should be tagged
>abandoned:building=house. If it cannot be recognised, it should be
>tagged ruins:building=yes
>Also note the taginfo counts here
>https://wiki.openstreetmap.org/wiki/Lifecycle_prefix (ruins: and
>ruined: are seldom used).
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/2177#issuecomment-716350516

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by rhhsm** [comment]
Hmm. the Nikolaikirche is indeed a tricky one. From the Wikipedia article I understand that only the tower is still standing, is being maintained and is used as a memorial, while the rubble of the rest of the church has been removed. I'd tag it with  man_made=tower, tower:type=bell_tower, building=yes (because of guidance here https://wiki.openstreetmap.org/wiki/Tag:tower:type%3Dbell_tower ) and historic=monument (to reflect its current use). Just after it was bombed it was a ruin, but since it's being maintained since then, it's not a ruin any more. It's being used, so not abandoned. And the rest of the church is gone, so not building=church.

You wrote above that "StreetComplete should enable mappers to map what they see." I fully agree: "Map what's on the ground" is one of the good practices of OSM. But there are many borderline cases where different mappers may choose different options, and in order to have some consistency in mapping, there are further guidelines, formed by consensus, that I think SC should follow. An example where SC follows an exception to "Map what's on the ground" is abbreviations on street signs https://wiki.openstreetmap.org/wiki/Names#Abbreviation_.28don.27t_do_it.29

I used the lifecycle prefix because the simple tag is discouraged https://wiki.openstreetmap.org/wiki/Key:abandoned:

If you disagree with the consensus, we should discuss it at the OSM wiki, not here.

**Comment by westnordost** [comment]
> I used the lifecycle prefix because the simple tag is discouraged https://wiki.openstreetmap.org/wiki/Key:abandoned:

> If you disagree with the consensus, we should discuss it at the OSM wiki, not here.

Right, I didn't notice it being discouraged as it is only mentioned in the continuous text.

**Comment by matkoniecz** [comment]
Though at the same time simple tag is mentioned as valid tagging solution, including `building` and `landuse=quarry` (though note that I was person who added this part in at least some pages, so do not count it as an independent support for my tagging ideas)

**Comment by peternewman** [comment]
At least in the UK, a lot of historic ruins will have an information board too, so you can see it's ruins and it used to be a manor house or a church or a watermill or whatever, even if it's just some wall outlines. Most other buildings are either still standing or have been knocked down and turned into houses etc.

**Comment by RiffLord** [comment]
I've been reading all of the comments and thinking about the issue, while also using StreetComplete daily. I do agree that it makes sense to tag buildings based on what they _are_ whenever possible, but there are cases when none of the available options feel appropriate (which is why I raised the issue in the first place).

> At least in the UK, a lot of historic ruins will have an information board too, so you can see it's ruins and it used to be a manor house or a church or a watermill or whatever, even if it's just some wall outlines. Most other buildings are either still standing or have been knocked down and turned into houses etc.

This is obviously very convenient for users in the UK as it could allow them to correctly tag these buildings. However, it's not the case in many other areas. The area where I live is full abandoned buildings, historic or otherwise, many of which can't be identified based solely on the appearance.

In order to tag them I could:

- Make an assumption based on the building's appearance. This approach increases the risk of error.
- Ask around to see if anybody living nearby remembers what a particular building was. This isn't necessarily a bad idea, but a SC user should have all the options available to tag a building themselves within the app.

Let me show a couple of examples:

![1Untitled](https://user-images.githubusercontent.com/47884610/97462161-bc22dc00-193e-11eb-8eff-d6e0da284f4a.jpg)

I happen to know that the above building used to be a restaurant, and as such I've tagged it so on StreetComplete. But just by looking at it there really is no way of knowing what it could have been.

<hr>

![2Untitled](https://user-images.githubusercontent.com/47884610/97462349-e4aad600-193e-11eb-9981-357ceac61206.jpg)

The above building looks to me as if it could have been a house, but I have no idea for sure. It could have just as easily been a small office building, or something else entirely. StreetComplete asked me what kind of building this was, and I'm not sure how to tag it.

<hr>

Both of these buildings have been abandoned for well over 15 years, and there's no sign that they will be demolished or restructured anytime soon. They are a defining part of the landscape just as much as the homes, stores and other buildings are.

To sum up, I agree that a building should be tagged based on what it is (or was), but my proposal is to add these _abandoned_ or _ruined_ categories for cases such as these, when the building's current state seems to be a defining factor more than what it originally was.

**Comment by matkoniecz** [comment]
> I happen to know that the above building used to be a restaurant, and as such I've tagged it so on StreetComplete. But just by looking at it there really is no way of knowing what it could have been.

By large windows/doors it seems clear to me that it is retail building (shop/restaurant) or maybe garage, how entrance looks like excludes garage (applies in my region, may not apply elsewhere)

> The above building looks to me as if it could have been a house, but I have no idea for sure. It could have just as easily been a small office building, or something else entirely. StreetComplete asked me what kind of building this was, and I'm not sure how to tag it.

That is a great example.

**Comment by rhhsm** [comment]
I think the best answer to give for these buildings is "Don't know". If you, as a local, can't figure out what kind of building it is, any user of your map data can't either. Maybe we should stop this discussion and go mapping :)

**Comment by westnordost** [comment]
@rhhsm but what can be figured out is that it is abandoned and that is a data point that is worth recording.

**Comment by RiffLord** [comment]
I believe that it would make sense to add these options to the quest, but to consider them special cases: while most of the options modify the `building` type tag, these should be implemented so as to add an `abandoned=yes`, `ruin=yes` or `historic=yes` tag, as suggested by @westnordost 

A separate quest would be annoyingly redundant, while simply answering _Don't Know_ is limiting, as it doesn't allow the user to insert some data which has value and which _can_ be inserted into the OSM ecosystem.

**Comment by westnordost** [comment]
So, you think these should be options in the "Other answers..." menu?

**Comment by RiffLord** [comment]
I would put them in the _Other_ category, the one with shed, bunker, service building, etc.

**Comment by westnordost** [comment]
That's fine by me

**Comment by smichel17** [comment]
I can imagine a future quest that asks, for abandoned buildings or ruins, if it is possible to determine the building's original purpose. I think that would address both sides here?

**Comment by peternewman** [comment]
> This is obviously very convenient for users in the UK as it could allow them to correctly tag these buildings. However, it's not the case in many other areas. The area where I live is full abandoned buildings, historic or otherwise, many of which can't be identified based solely on the appearance.

Perhaps I was overly optimistic, or primarily concentrating on historic ruins @RiffLord . I agree I'd certainly struggle with the specifics of your second example.

**Comment by rhhsm** [comment]
> 
> 
> > I used the lifecycle prefix because the simple tag is discouraged https://wiki.openstreetmap.org/wiki/Key:abandoned:
> 
> > If you disagree with the consensus, we should discuss it at the OSM wiki, not here.
> 
> Right, I didn't notice it being discouraged as it is only mentioned in the continuous text.

You may want to wait on deciding on this until this https://github.com/openstreetmap/iD/issues/8051 is settled.

**Comment by matkoniecz** [comment]
> You may want to wait on deciding on this until this openstreetmap/iD#8051 is settled.

This is internal iD rendering issue with no relevance for tagging.

**Comment by RiffLord** [comment]
Forgive me for cluttering everyone's mailboxes, but I'm unsure where to ask this: I've implemented the changes on a separate branch and will be ready to make a pull request just as soon as I translate the string resources to the languages which I'm fluent in.

Should I merge the changes to the master branch before the pull request or will the repo owner do this afterwards?

**Comment by westnordost** [comment]
> just as soon as I translate the string resources to the languages which I'm fluent in.

You don't need and should not do that. Translations happen on POEditor, anything done in the repo will be overwritten by hte translations pulled from htere.

> Should I merge the changes to the master branch before the pull request or will the repo owner do this afterwards?

Not sure what you mean. The PR should be mergeable, i.e. there should not be any conflicts with the current master.

**Comment by rhhsm** [comment]
> 
> 
> > You may want to wait on deciding on this until this [openstreetmap/iD#8051](https://github.com/openstreetmap/iD/issues/8051) is settled.
> 
> This is an internal iD rendering issue with no relevance for tagging and not a place for OSM tagging discussions. Also, it is settled - closed and fixed.

Thanks for your comments there and your edits on OSM wiki. For me it is important that SC follows consensus, not so much what that consensus is. Here it seems to depend mostly on software coding issues that I'm not qualified to comment on.

**Comment by RiffLord** [comment]
> You don't need and should not do that. Translations happen on POEditor, anything done in the repo will be overwritten by hte translations pulled from htere.

Alright, I won't do anything about that then.

> Not sure what you mean. The PR should be mergeable, i.e. there should not be any conflicts with the current master.

I'm still quite new to the _foss_ way of doing things, I just wanted to know who should perform the merge in this case.

**Comment by westnordost** [comment]
Yes, you do the changes on a branch in a fork of this repository. Then, you create a new pull request (also known as merge request) to request merging your changes into this repository. This pull request is managed like an issue ticket, I can review it, comment on changes, request changes etc. and also finally merge it.

**Comment by peternewman** [comment]
> I'm still quite new to the _foss_ way of doing things, I just wanted to know who should perform the merge in this case.

Only people with maintainer rights (i.e. probably not you) will actually be allowed to merge it into this repo @RiffLord .

You can experiment/compare this on your own repo, if you or I open a PR into your version of it from one branch to another, you will see a merge button you can click to action the merge.
